ui/src/components/schemas/SchemaDetails.tsx

```diff
@@ -0,0 +1,340 @@
+                status: "failed",
+              });
+            }
+          })
+          .catch((error) => {
+            setJsonLdType({
+              error: extractError(error),
+              status: "failed",
+            });
+          });
+      })
+      .catch((error) => {
+        setSchema({
+          error: extractError(error),
+          status: "failed",
+        });
+      });
+  }, []);
+
+  const onGetApiSchema = useCallback(
+    async (signal: AbortSignal) => {
+      if (schemaID) {
+        setApiSchema({ status: "loading" });
+
+        const response = await getSchema({
+          env,
+          schemaID,
+          signal,
+        });
+
+        if (response.isSuccessful) {
+          setApiSchema({ data: response.data, status: "successful" });
+          fetchSchemaFromUrl(response.data);
+        } else {
+          if (!isAbortedError(response.error)) {
+            setApiSchema({ error: response.error, status: "failed" });
+          }
+        }
+      }
+    },
+    [env, fetchSchemaFromUrl, schemaID]
+  );
+
+  useEffect(() => {
+    if (schemaID) {
+      const { aborter } = makeRequestAbortable(onGetApiSchema);
+      return aborter;
+    }
+    return;
+  }, [onGetApiSchema, schemaID]);
+
+  const jsonLdTypeErrorToString = (error: string | z.ZodError) =>
+    error instanceof z.ZodError
+      ? [
+          "An error occurred while trying to parse the JSON LD Type referenced in this schema:",
+          ...processZodError(error).map((e) => `"${e}"`),
+          "Please provide a schema with a valid JSON LD Type.",
+        ].join("\n")
+      : `An error occurred while downloading the JSON LD Type referenced in this schema:\n"${error}"\nPlease try again.`;
+
+  const schemaErrorToString = (error: string | z.ZodError) =>
+    error instanceof z.ZodError
+      ? [
+          "An error occurred while trying to parse this schema:",
+          ...processZodError(error).map((e) => `"${e}"`),
+          "Please provide a valid JSON Schema.",
+        ].join("\n")
+      : `An error occurred while downloading this schema:\n"${error}"\nPlease try again.`;
+
+  const loading =
+    isAsyncTaskStarting(apiSchema) ||
+    isAsyncTaskStarting(schema) ||
+    isAsyncTaskStarting(jsonLdType);
+
+  const errorMsg =
+    apiSchema.status === "failed"
+      ? apiSchema.error.message
+      : schema.status === "failed"
+      ? schemaErrorToString(schema.error)
+      : jsonLdType.status === "failed"
+      ? jsonLdTypeErrorToString(jsonLdType.error)
+      : undefined;
+
+  return (
+    <SiderLayoutContent
+      description="Schema details include a hash, schema URL and attributes. Schema can be viewed in a formatted way as well as, LD Context and schema."
+      showBackButton
+      showDivider
+      title="Schema details"
+    >
+      {(() => {
+        if (
+          apiSchema.status === "failed" ||
+          schema.status === "failed" ||
+          jsonLdType.status === "failed"
```

Comment:
you can do the same that `loading` with `hasAsyncTaskFailed`

Reply:
Yep I realize this existed in the last PR I opened 🤣

Reply:
Done!